From: Mingming Cao <cmm@us.ibm.com>

Turn on extents feature by default in ext4 filesystem, to get wider testing of extents feature in ext4dev. User could use -o noextents to turn it off. For productive ext4 filesystem, extents feature will be enabled at mkfs time, and only mounted fs with extents if the fs has this flag turned on.

Signed-off-by: Mingming Cao <cmm@us.ibm.com>

---
 fs/ext4/super.c |    6 ++++++
 1 file changed, 6 insertions(+)

Index: linux-2.6.22/fs/ext4/super.c
===================================================================
--- linux-2.6.22.orig/fs/ext4/super.c	2007-07-16 00:56:54.000000000 -0700
+++ linux-2.6.22/fs/ext4/super.c	2007-07-16 00:57:06.000000000 -0700
@@ -1546,6 +1546,12 @@ static int ext4_fill_super (struct super
 
 	set_opt(sbi->s_mount_opt, RESERVATION);
 
+	/*
+	 * turn on extents feature by default in ext4 filesystem
+	 * User -o noextents to turn it off
+	 */
+	set_opt(sbi->s_mount_opt, EXTENTS);
+
 	if (!parse_options ((char *) data, sb, &journal_inum, &journal_devnum,
 			    NULL, 0))
 		goto failed_mount;
